packages/transport-amqp/test/error-handling.spec.ts

```diff
@@ -0,0 +1,132 @@
+// import { HttpStatusError, Error as CommonError } from 'common-errors'
+// import { route as Proxy } from '@microfleet/amqp-coffee/test/proxy.js'
+// import ld from 'lodash'
+// import stringify from 'json-stringify-safe'
+// import sinon from 'sinon'
+import assert from 'assert'
+// import microtime from 'microtime'
+// import { promisify } from 'util'
+// import { gzip as _gzip } from 'zlib'
+// import { setTimeout } from 'timers/promises'
+// import { timesLimit } from 'async'
+const debug = require('debug')('amqp:test')
+
+// require module
+import {
+  AMQPTransport,
+  // connect,
+  // jsonSerializer,
+  // jsonDeserializer,
+  // connect,
+  // multiConnect,
+  // ExtendedMessageProperties,
+  // ResponseHandler,
+  // kReplyHeaders,
+} from '../src'
+// import { toMiliseconds } from '../src/utils/latency'
+import {
+  ConnectionState,
+  // ExchangeBindOptions,
+  // Message
+} from '@microfleet/amqp-coffee'
+// import { HttpStatusError } from "common-errors";
+// import { QueueArguments } from "@microfleet/transport-amqp";
+// import { v4 } from 'uuid'
+
+// eslint-disable-next-line @typescript-eslint/no-var-requires
+// const debug = require('debug')('amqp')
+// const gzip = promisify(_gzip)
+
+describe('AMQPTransport', function AMQPTransportTestSuite() {
+
+  this.timeout(600_000)
+
+  const RABBITMQ_HOST = process.env.RABBITMQ_PORT_5672_TCP_ADDR || 'localhost'
+  const RABBITMQ_PORT = +(process.env.RABBITMQ_PORT_5672_TCP_PORT || 5672)
+
+  const configuration = {
+    exchange: 'test-exchange',
+    connection: {
+      host: RABBITMQ_HOST,
+      port: RABBITMQ_PORT,
+    },
+  }
+
+  let amqp: AMQPTransport
+
+  it('is able to connect to rabbitmq', async () => {
+    amqp = new AMQPTransport(configuration)
+    await amqp.connect()
+    assert.equal(amqp.state, ConnectionState.open)
+  })
+
+  it('should be able to publish existing exchange', async () => {
+    for(let i=0; i<20; i++) {
+      debug(`creating channel ${i}`)
+      try {
+        await amqp.publish("test", { "foo": "bar" }, { confirm: true })
+      } catch (err) {
+        // console.log(err)
+      }
+      // await new Promise(h => setTimeout(h, 1_000))
+    }
+  })
+
+  it('should throw 404 on publish to non-existing exchange', async () => {
+    for(let i=0; i<5; i++) {
+      debug(`creating channel ${i}`)
+      try {
+        await amqp.publish("test", { "foo": "bar" }, { confirm: true, exchange: "non-existing" })
+      } catch (err) {
+        // console.log(err)
+      }
+      // await new Promise(h => setTimeout(h, 1_000))
+    }
+  })
+
+  it('should be safe to delete non-existing queue', async () => {
+    const { queue: queue1 } = await amqp.createQueue({ queue: 'redeclared-queue' })
+    const { queue: queue2 } = await amqp.createQueue({ queue: 'redeclared-queue' })
+    await queue1.delete()
+    await queue2.delete()
```

Comment:
add cases where queue isnt empty and we try to delete it for 406 error
also add queue.declare with passive: true where queue doesnt exist (it would also throw)